src/main/scala/chisel3/stage/ChiselStage.scala

```diff
@@ -22,17 +22,7 @@ class ChiselStage extends Stage {
 
   val shell: Shell = new Shell("chisel") with ChiselCli with FirrtlCli
 
-  val targets: Seq[Dependency[Phase]] =
-    Seq( Dependency[chisel3.stage.phases.Checks],
-         Dependency[chisel3.stage.phases.AddImplicitOutputFile],
-         Dependency[chisel3.stage.phases.AddImplicitOutputAnnotationFile],
-         Dependency[chisel3.stage.phases.MaybeAspectPhase],
-         Dependency[chisel3.stage.phases.Convert],
-         Dependency[chisel3.stage.phases.MaybeFirrtlStage] )
-
-  final lazy val phaseManager = new PhaseManager(targets) {
-    override val wrappers = Seq( (a: Phase) => DeletedWrapper(a) )
-  }
+  final lazy val phaseManager = new ChiselPhase
```

Comment:
I don't think we should remove this API without first deprecating. It is used by rocket-chip after all: https://github.com/chipsalliance/rocket-chip/blob/d71c5a674275676b9c870a20cb0a42cd039a4115/src/main/scala/system/RocketChipStageGenerator.scala#L15

I dunno how many other users there might be, but I'd say preserve it by keeping `phaseManager` the same and then implement targets as `def targets: Seq[Dependency[Phase]] = (new ChiselPhase).targets`, then it can still be overridden to affect `phaseManager`, feel free to deprecate it though, instead suggesting composition I assume?

Reply:
Good catch. Updated in the rebase. There's annoyingly no good Scala 2.11 compatible way to deprecate this. `@deprecatedOverriding` is what we want, but it's a 2.12 API addition. (And just using `@deprecated` produces no warnings if you override it.)

That said, I feel like there are some warts with the `Stage` API that can be better handled with a flat structure. So, I wouldn't sweat this and would rather put more thought into how to better express this flat structure (which would likely be a larger update to the `Stage` API.)